### ISSUE TITLE
On time route incorrectly showing as delayed/early color

### ISSUE THREAD

**Issue by bbodenmiller** [OneBusAway/onebusaway-android#363]
Running latest alpha build from play store I'm getting this sometimes:

![screenshot_20151115-215754](https://cloud.githubusercontent.com/assets/1192780/11175177/2f2c5d68-8be4-11e5-94e3-66a1902acc70.png)
![screenshot_20151115-215829](https://cloud.githubusercontent.com/assets/1192780/11175191/5298be86-8be4-11e5-913a-e46adb441ce4.png)

**Comment by bbodenmiller** [comment]
And now this:

![screenshot_20151115-220234](https://cloud.githubusercontent.com/assets/1192780/11175226/a0d46532-8be4-11e5-9664-6b66e6393b41.png)

**Comment by barbeau** [comment]
I've seen this myself before as well, although I haven't seen it lately.  I recall seeing this in past production versions (v1.x) on Google Play as well, so I think its a bug that goes way back.  I did a quick look through the code at one point and didn't see anything obvious as to why this happens, but I'll look again.

**Comment by barbeau** [comment]
@bbodenmiller The next time you see this, can you see if hitting the "Refresh" button fixes the colors?

**Comment by bbodenmiller** [comment]
Will do.

**Comment by barbeau** [comment]
After a closer look, I think this might be a rounding issue in the `ArrivalInfo()` constructor.  The OBA REST API returns times as milliseconds (ms) since the unix epoch.  For the color, the ms values for estimated and scheduled times are used to determine if its early/ontime/late in `computeColor()`.  For the ETA and status text, the ms values for estimated and scheduled times are converted to minutes, and then those minutes values are used to determine early/ontime/late in `computeStatusLabel()`.

**Comment by barbeau** [comment]
The rounding error could explain why this only happens occasionally, too.  For example, in Tampa the delay values are all calculated at the minute precision - they are expressed in seconds in the GTFS-rt feed, but always "60", "120", etc.  This issue would only surface if the scheduled arrival time wasn't on the minute, or if the delay value wasn't a multiple of 60 seconds (or an absolute time was supplied in the source data for predicted arrival that wasn't on the minute).  These occurrences might be more frequent in some regions than others, depending on the GTFS and GTFS-rt (or other source format) data.

**Comment by barbeau** [comment]
Yes, seems like in Puget Sound the "5 - Downtown Seattle Greenwood" that @bbodenmiller took screenshots of above regularly sees times not rounded to the minute that make this bug visible.  I can't reproduce in Tampa, but I can reproduce if I look at stops on that route (I just looked at `stop_code` 5990 and saw it).  I just changed `ArrivalInfo` to calculate the color based on the values rounded to the minute instead of ms (just like the status text), and it seems to resolve this issue.  I'll push a fix shortly.
